trie/proof.go

```diff
@@ -112,6 +112,9 @@ func VerifyProof(rootHash common.Hash, key []byte, proofDb ethdb.Reader) (value
 		if buf == nil {
 			return nil, i, fmt.Errorf("proof node %d (hash %064x) missing", i, wantHash)
 		}
+		if bytes.Compare(crypto.Keccak256(buf), wantHash[:]) != 0 {
```

Comment:
Tiny request, could we create 1 hasher and reuse it? This `crypto.Keccak256` will create and discard one for every trie node. Otherwise very nice catch!

Reply:
Ah, also pleas use `bytes.Equal`, this will fail the linter.

Reply:
Updated.